torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -436,9 +446,12 @@ struct GraphFuser {
       // handle inputs in reverse topological order as well...
       // otherwise in f(a,a+b) it will appear a is used twice if we consider
       // the f-a fusion before the f-(a+b) fusion first.
-      value_list inputs = consumer->inputs();
-      for(auto i : inputs) {
-        JIT_ASSERT(topological_index.count(i->node()) > 0);
+      value_list inputs;
+      for(auto i : consumer->inputs()) {
+        if (i->node()->owningBlock() == block) {
+          inputs.push_back(i);
+          JIT_ASSERT(topological_index.count(i->node()) > 0);
+        }
```

Comment:
Was this necessary? I think `isFusable` was supposed to fail for nodes in different blocks, so it should have been fine without filtering the inputs

Reply:
This was failing in James' beam search code. The topological order of the nodes is only defined for the current block under consider and this assert (and the later sort) fails before the isFusible check. The isFusible check is still necessary as well due to Chunking optimization. If we see more errors later, we should consider taking a different approach to where we check for suitable fusion edges.